### ISSUE TITLE
[Bug]: InconsistentAnalysisException when generating model

### ISSUE THREAD

**Issue by Navil** [realm/realm-dart#879]
### What happened?

When running
`flutter pub run realm generate`

the command fails and I get "[SEVERE]" logs. This happens on two independend machines. Restarting the computer occassionally helps. The other files/steps run as usual. This error did not happen in the past and did also occur before upgrading Flutter to 3.3.0.

### Repro steps

`flutter pub run realm generate`

### Version

Flutter (Channel stable, 3.3.0, on macOS 12.5.1 21G83 darwin-x64, locale en-GB)

### What Realm SDK flavor are you using?

MongoDB Atlas (i.e. Sync, auth, functions)

### What type of application is this?

Flutter Application

### Client OS and version

0.4.0+beta

### Relevant log output

```shell
[INFO] realm:realm_generator on lib/shared/util/date_helper.dart:[generate (0)] completed, took 3ms
[SEVERE] realm:realm_generator on lib/shared/util/date_helper.dart:

InconsistentAnalysisException: Requested result might be inconsistent with previously returned results

[INFO] 29.3s elapsed, 1/16 actions completed.
[INFO] realm:realm_generator on lib/shared/util/country_select.dart:[generate (0)] completed, took 129ms
[INFO] realm:realm_generator on lib/shared/util/custom_alert.dart:[generate (0)] completed, took 57ms
[INFO] realm:realm_generator on lib/shared/util/custom_route.dart:[generate (0)] completed, took 112μs
[SEVERE] realm:realm_generator on lib/shared/util/custom_route.dart:

InconsistentAnalysisException: Requested result might be inconsistent with previously returned results

[INFO] 31.5s elapsed, 4/20 actions completed.
[INFO] realm:realm_generator on lib/firebase_options.dart:[generate (0)] completed, took 61μs
[SEVERE] realm:realm_generator on lib/firebase_options.dart:

InconsistentAnalysisException: Requested result might be inconsistent with previously returned results

[INFO] 33.3s elapsed, 5/21 actions completed.
```

**Comment by nielsenko** [comment]
Hi Thomas

If you check the `pubspec.lock` on the two machines in question. What `analyzer` package are they using?

**Comment by Navil** [comment]
> Hi Thomas
> 
> If you check the `pubspec.lock` on the two machines in question. What `analyzer` package are they using?

```
analyzer:
    dependency: transitive
    description:
      name: analyzer
      url: "https://pub.dartlang.org"
    source: hosted
    version: "4.6.0"
```

**Comment by nielsenko** [comment]
We see this issue on our CI as well from time to time. Fx:
https://github.com/realm/realm-dart/runs/8274059582?check_suite_focus=true,
but I have yet to re-create it on my own machine.

It seems less frequent after upgrading `analyzer` to 4.7.0, so you may try upgrading, but it still occurs. We need to investigate more. so If you have any more observations to share, they are much appreciated.

**Comment by Navil** [comment]
I just updated to 4.7.0 and experienced the same issue on my first try. But I can give an update if a few days if I feel that it got less. Tried with the --clean flag. Same issue. But the files however generate, so I can "ignore" the error for now.

**Comment by nielsenko** [comment]
@Navil Fix just released with [0.5.0+beta](https://pub.dev/packages/realm/versions/0.5.0+beta)
